### PR TITLE
kickstart models and items controller

### PR DESCRIPTION
**Pull request by stshayna**
@DenzelBraithwaite Please take a look of the core models and items controller created. Thanks!
Models that haven't been created: messages, chatrooms and feedbacks.

Screenshot 2022-05-18 at 8 50 32 PM
.

### CODE REVIEW

File: app/models/allergen.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# == Schema Information
+#
+# Table name: allergens
+#
+#  id         :bigint           not null, primary key
+#  name       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+class Allergen < ApplicationRecord
```
Comment (DenzelBraithwaite) [inline app/models/allergen.rb:10]: We'll need to validate the name presence for this after.

File: app/models/diet.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# == Schema Information
+#
+# Table name: diets
+#
+#  id         :bigint           not null, primary key
+#  name       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+class Diet < ApplicationRecord
```
Comment (DenzelBraithwaite) [inline app/models/diet.rb:10]: We'll need to validate the name presence for this after.

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class PagesController < ApplicationController
-  skip_before_action :authenticate_user!, only: [ :home ]
+  skip_before_action :authenticate_user!, only: [:home]
```
Comment (DenzelBraithwaite) [inline app/controllers/pages_controller.rb:2]: Tested, good.

File: Gemfile [hunk 4]
```diff
@@ -40,6 +40,8 @@ gem 'simple_form', github: 'heartcombo/simple_form'
 group :development, :test do  gem 'pry-byebug'
   gem 'pry-rails'
   gem 'dotenv-rails'
+  # Get a summary of the Active Record schema in relevant files, e.g. models
+  gem 'annotate'
```
Comment (DenzelBraithwaite) [inline Gemfile:44]: Neat gem

File: app/models/item.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+#  id              :bigint           not null, primary key
+#  description     :text
+#  expiration_date :date
+#  status          :integer          default("available")
+#  type            :string
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  user_id         :bigint           not null
+#
+class Item < ApplicationRecord
```
Comment (DenzelBraithwaite) [inline app/models/item.rb:14]: item.allers/user/diets all working

Comment (DenzelBraithwaite) [reply app/models/item.rb:14]: allergens*

File: app/models/items_allergen.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+#
+# Table name: items_allergens
+#
+#  id          :bigint           not null, primary key
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#  allergen_id :bigint           not null
+#  item_id     :bigint           not null
+#
+class ItemsAllergen < ApplicationRecord
```
Comment (DenzelBraithwaite) [inline app/models/items_allergen.rb:11]: tested, both working. (.item or .allergen)

File: app/models/items_diet.rb [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+#
+# Table name: items_diets
+#
+#  id         :bigint           not null, primary key
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#  diet_id    :bigint           not null
+#  item_id    :bigint           not null
+#
+class ItemsDiet < ApplicationRecord
```
Comment (DenzelBraithwaite) [inline app/models/items_diet.rb:11]: Tested, good

File: app/views/devise/registrations/new.html.erb [hunk 8]
```diff
@@ -1,25 +1,33 @@
...
-                required: true,
-                input_html: { autocomplete: "new-password" } %>
-  </div>
+        <div class="form-inputs">
+          <%= f.input :username, autofocus: true %>
+          <%= f.input :email,
+                      required: true,
+                      autofocus: true,
+                      input_html: { autocomplete: "email" }%>
+          <%= f.input :address, autofocus: true %>
```
Comment (DenzelBraithwaite) [inline app/views/devise/registrations/new.html.erb:15]: Not sure if having 2 autofocus will cause an issue, I'm unfamiliar with the code.

File: app/models/transaction.rb [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+# Table name: transactions
+#
+#  id          :bigint           not null, primary key
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#  giver_id    :bigint           not null
+#  item_id     :bigint           not null
+#  receiver_id :bigint           not null
+#
+class Transaction < ApplicationRecord
```
Comment (DenzelBraithwaite) [inline app/models/transaction.rb:12]: all good

File: app/models/user.rb [hunk 10]
```diff
@@ -1,6 +1,30 @@
...
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  has_many :items, dependent: :destroy
```
Comment (DenzelBraithwaite) [inline app/models/user.rb:22]: tested, good

File: db/migrate/20220518011803_add_columns_to_user.rb [hunk 11]
```diff
@@ -0,0 +1,6 @@
+class AddColumnsToUser < ActiveRecord::Migration[6.1]
+  def change
```
Comment (DenzelBraithwaite) [inline db/migrate/20220518011803_add_columns_to_user.rb:2]: yep they were added successfully.

**Review by DenzelBraithwaite — APPROVED** [review]
A lot of work done here, try to split up the work a bit more but other than that, great work. Everything I tested and reviewed seemed to work well. There were a few things I was unsure of but I'm going to assume you did your homework. Merging!

File: app/models/allergen.rb (hunk 1 above)
Comment (stshayna) [reply app/models/allergen.rb:10]: added validation (presence and uniqueness) today, and also added validations on other models

File: app/views/devise/registrations/new.html.erb (hunk 8 above)
Comment (stshayna) [reply app/views/devise/registrations/new.html.erb:15]: Checked in http://localhost:3000/users/sign_up and it was ok!

**Pull request merged**